File: common/src/main/java/com/goexp/common/util/Gzip.java
package com.goexp.common.util;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.nio.ByteBuffer;
import java.nio.charset.Charset;
import java.util.Objects;
import java.util.zip.GZIPInputStream;

public class Gzip {
    public static byte[] unGzip(byte[] compressedBytes) {
        try (var s = new GZIPInputStream(new ByteArrayInputStream(compressedBytes))) {
            return s.readAllBytes();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return null;
    }

    public static String decode(byte[] compressedBytes, Charset charset) {

        Objects.requireNonNull(compressedBytes);
        Objects.requireNonNull(charset);

        byte[] bytes = unGzip(compressedBytes);
        Objects.requireNonNull(bytes, "Unzip error");

        return charset.decode(ByteBuffer.wrap(bytes)).toString();
    }
}

File: common/src/main/scala/com/goexp/common/util/Gzip.scala
package com.goexp.common.util

import java.io.{ByteArrayInputStream, IOException}
import java.nio.ByteBuffer
import java.nio.charset.Charset
import java.util.Objects
import java.util.zip.GZIPInputStream

object Gzip {

  implicit class GzipSupport(compressedBytes: Array[Byte]) {
    def unGzip() = {
      Gzip.unGzip(compressedBytes)
    }
  }

  private def unGzip(compressedBytes: Array[Byte]): Array[Byte] = {
    val s = new GZIPInputStream(new ByteArrayInputStream(compressedBytes))
    try return s.readAllBytes
    catch {
      case e: IOException =>
        e.printStackTrace()
    } finally if (s != null) s.close()
    null
  }

  def decode(compressedBytes: Array[Byte], charset: Charset): String = {
    Objects.requireNonNull(compressedBytes)
    Objects.requireNonNull(charset)
    val bytes = unGzip(compressedBytes)
    Objects.requireNonNull(bytes, "Unzip error")
    charset.decode(ByteBuffer.wrap(bytes)).toString
  }
}